### PR TITLE
thresholds layout & logic

### PR DESCRIPTION
**Pull request by elevchyt**
Adds a new "Thresholds" button next to the advanced search button which allows users to set minimum and/or maximum thresholds of the current OrderBy unit in order to limit the results to those who fill the desired thresholds' criteria.

closes https://github.com/adsviewer/turboviewer/issues/655

### CODE REVIEW

File: apps/web/package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
     "react": "19.0.0-rc-02c0e824-20241028",
     "react-dom": "19.0.0-rc-02c0e824-20241028",
     "react-hook-form": "^7.53.2",
+    "react-icons": "^5.4.0",
```
Comment (trixobird) [inline apps/web/package.json:78]: do we use this?

File: apps/web/src/components/thresholds/thresholds.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
+import { ActionIcon, Modal, Tooltip, Text, NumberInput, Flex, Button } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { IconAdjustmentsHorizontal, IconCancel } from '@tabler/icons-react';
```
Comment (trixobird) [inline apps/web/src/components/thresholds/thresholds.tsx:3]: This icon does not transmit intention. Looks more like filters

File: apps/web/src/components/thresholds/thresholds.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+          <NumberInput
+            label={tGeneric('minimum')}
+            placeholder={tGeneric('none')}
+            thousandSeparator=","
+            allowNegative={false}
+            value={minValue ?? ''}
+            onChange={handleMinValueChange}
+            min={1}
+          />
+          <NumberInput
```
Comment (trixobird) [inline apps/web/src/components/thresholds/thresholds.tsx:130]: https://www.loom.com/share/a0fbbcd60ffe48f3b69a2feb08aa7045?sid=6bd74d33-2830-4c41-b5e0-87a7bf1a5365

**Review by trixobird — APPROVED** [review]

File: apps/web/src/components/thresholds/thresholds.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    props.startTransition(() => {
+      router.replace(finalURL, { scroll: false });
+      close();
+    });
+  };
+
+  const hasActiveThresholds = (): boolean => {
+    if (searchParams.get(urlKeys.minThreshold) ?? searchParams.get(urlKeys.maxThreshold)) return true;
+    return false;
+  };
```
[the remark this answers is not in the document]
Comment (elevchyt) [reply apps/web/src/components/thresholds/thresholds.tsx:90]: damn i was sure there was a better way to do this..

File: apps/web/src/components/thresholds/thresholds.tsx (hunk 2 above)
Comment (elevchyt) [reply apps/web/src/components/thresholds/thresholds.tsx:3]: Yeah I searched a lot and it was the best one I could find

File: apps/web/src/components/thresholds/thresholds.tsx (hunk 3 above)
Comment (elevchyt) [reply apps/web/src/components/thresholds/thresholds.tsx:128]: king of testing

File: apps/web/package.json (hunk 1 above)
Comment (elevchyt) [reply apps/web/package.json:78]: Eh not really, but I added it in hopes of finding a better icon. Will definitely use it at some point in the near future so I'd like to keep it if you don't mind.

**Pull request merged**